dedupe/convenience.py

```diff
@@ -24,55 +24,88 @@ def consoleLabel(deduper): # pragma: no cover
     '''
 
     finished = False
+    use_previous = False
     fields = unique(field.field
                     for field
                     in deduper.data_model.primary_fields)
 
+    buffer_len = 1 # Max number of previous operations
+    record_label_buffer = []
+    uncertain_pairs = []
 
-    while not finished :
-        n_match, n_distinct = (len(deduper.training_pairs['match']),
-                               len(deduper.training_pairs['distinct']))
-
-        uncertain_pairs = deduper.uncertainPairs()
-
-        examples = {'distinct' : [], 'match' : []}
-
-        for record_pair in uncertain_pairs:
-            label = ''
-            labeled = False
-
-            for pair in record_pair:
-                for field in fields:
-                    line = "%s : %s" % (field, pair[field])
-                    print(line, file=sys.stderr)
-                print(file=sys.stderr) 
-
-            print("{0}/10 positive, {1}/10 negative".format(n_match, n_distinct),
-                  file=sys.stderr)
-            print('Do these records refer to the same thing?', file=sys.stderr)
-            valid_response = False
-            while not valid_response:
+    while not finished :       
+        if len(record_label_buffer) and use_previous:
+            uncertain_pairs.append(record_label_buffer[0][0]) # append previous record_pair
+            record_label_buffer = record_label_buffer[1:] # remove previous pair from buffer
```

Comment:
These lines correspond to the situation where `previous` was chosen so we want to add the last labeled element to the list of `uncertain_pairs`. Since I append at front of `record_label_buffer`, in this case I want to pop(0) to retrieve the latest element.